client/js/table-view.js

```diff
@@ -72,22 +73,63 @@ class TableView {
     this.sheetBodyEl.appendChild(fragment);
   }
 
+  renderSumRow(){
+    const tr = createTR();
+    for(let col = 0; col < this.model.numCols; col++){
+      const position = {col: col};
+      const value = this.model.getValue(position);
+      const td = createTD(value);
+      tr.appendChild(td);
+      td.className = 'sum-row';
+    }
+    this.sheetBodyEl.appendChild(tr);
+
+    this.sumRowEl = document.querySelectorAll('.sum-row');
+
+  }
+
   attachEventHandlers(){
     this.sheetBodyEl.addEventListener('click', this.handleSheetClick.bind(this));
+    this.formulaBarEl.addEventListener('keyup', this.handleSumColumnUpdate.bind(this));
     this.formulaBarEl.addEventListener('keyup', this.handleFormulaBarChange.bind(this));
   }
 
   handleFormulaBarChange(evt){
     const value = this.formulaBarEl.value;
     this.model.setValue(this.currentCellLocation, value);
     this.renderTableBody();
+    this.renderSumRow();
+  }
+
+  handleSumColumnUpdate(evt){
```

Comment:
This function ended up checking for a number of things. Could you split it up with a helper function?